packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -196,19 +196,19 @@ export function pasteHandler( {
 		}
 	}
 
-	if ( mode === 'INLINE' ) {
-		return filterInlineHTML( HTML );
-	}
-
-	// An array of HTML strings and block objects. The blocks replace matched
-	// shortcodes.
+	// An array of HTML strings and block objects.
+	// The blocks replace matched shortcodes.
 	const pieces = shortcodeConverter( HTML );
 
 	// The call to shortcodeConverter will always return more than one element
 	// if shortcodes are matched. The reason is when shortcodes are matched
 	// empty HTML strings are included.
 	const hasShortcodes = pieces.length > 1;
 
+	if ( mode === 'INLINE' && ! hasShortcodes ) {
```

Comment:
It seems to me that this should be fine because shortcodes are converted to blocks, which can't be inline. cc @ellatrix

Reply:
I've only checked the [default shortcodes](https://codex.wordpress.org/Shortcode_API#Default_Shortcodes), though. Perhaps, there is some shortcode that could be converted into inline formats. However, I don't see that's possible to do right now with the transformations API.

Reply:
Yeah, I think it's fine if all the tests pass.

Reply:
I disagree, I think it's fairly important to respect any third-party **inline** shortcodes. In my experience, the use cases are many and contemplate anything from inline formatting to marking positions in the document. (The question of whether shortcodes remain the right tool is not ours to ask here!)

> It seems to me that this should be fine because shortcodes are converted to blocks, which can't be inline

Because even shortcodes that don't have a known associated block type will be picked up by the catch-all Shortcode block, we would be suddenly forcing all shortcodes to be block-like. One _could_ let the editor differentiate between shortcodes with a known block type (which can never be inline) and unknown shortcodes (which may or not be inline), but I'd honestly stay away from implementing that. It would introduce a lot of complexity for little benefit.

Instead, it would be more interesting if the block editor were smart enough to decide when a pasted shortcode is meant to be inline or not. If it is, the editor should abstain from trying to make it a block. Concretely, these are some behaviours to consider:

1. Start a new line (i.e. a blank Paragraph) and paste a shortcode → Add a block for that shortcode.
2. Place the caret in the middle of some text and paste a shortcode → Insert the shortcode text verbatim, no new blocks.
3. Place the caret at the end of a paragraph of text and paste a shortcode → …? (up to us to decide)